ark/jsonschema/scope.ts

```diff
@@ -0,0 +1,78 @@
+import { scope } from "arktype"
+
+const $ = scope({
+	AnyKeywords: {
+		"const?": "unknown",
+		"enum?": "unknown[]"
+	},
+	CompositionKeywords: {
+		"allOf?": "Schema[]",
+		"anyOf?": "Schema[]",
+		"oneOf?": "Schema[]",
+		"not?": "Schema"
+	},
+	TypeWithNoKeywords: { type: "'boolean'|'null'" },
+	TypeWithKeywords: "ArraySchema|NumberSchema|ObjectSchema|StringSchema",
+	// NB: For sake of simplicitly, at runtime it's assumed that
+	// whatever we're parsing is valid JSON since it will be 99% of the time.
+	// This decision may be changed later, e.g. when a built-in JSON type exists in AT.
+	Json: "unknown",
+	"#BaseSchema":
+		// NB: `true` means "accept an valid JSON"; `false` means "reject everything".
+		"boolean|TypeWithNoKeywords|TypeWithKeywords|AnyKeywords|CompositionKeywords",
+	Schema: "BaseSchema|BaseSchema[]",
+	ArraySchema: {
+		"additionalItems?": "Schema",
+		"contains?": "Schema",
+		// JSON Schema states that if 'items' is not present, then treat as an empty schema (i.e. accept any valid JSON)
+		"items?": "Schema|Schema[]",
+		"maxItems?": "number.integer>=0",
+		"minItems?": "number.integer>=0",
+		type: "'array'",
+		"uniqueItems?": "boolean"
+	},
+	NumberSchema: {
+		// NB: Technically 'exclusiveMaximum' and 'exclusiveMinimum' are mutually exclusive with 'maximum' and 'minimum', respectively,
+		// which is reflected at runtime but it's not worth the performance cost to validate this statically.
+		"exclusiveMaximum?": "number",
+		"exclusiveMinimum?": "number",
+		"maximum?": "number",
+		"minimum?": "number",
+		// NB: JSON Schema allows decimal multipleOf, but ArkType only supports integer.
+		"multipleOf?": "number.integer",
+		type: "'number'|'integer'"
+	},
+	ObjectSchema: {
+		"additionalProperties?": "Schema",
+		"maxProperties?": "number.integer>=0",
+		"minProperties?": "number.integer>=0",
+		"patternProperties?": { "[string]": "Schema" },
+		// NB: Technically 'properties' is required when 'required' is present,
+		// which is reflected at runtime but it's not worth the performance cost to validate this statically.
+		"properties?": { "[string]": "Schema" },
+		"propertyNames?": "Schema",
+		"required?": "string[]",
+		type: "'object'"
+	},
+	StringSchema: {
+		"maxLength?": "number.integer>=0",
+		"minLength?": "number.integer>=0",
+		"pattern?": "RegExp | string",
+		type: "'string'"
+	}
+})
+export const JsonSchema = $.export()
+
+export declare namespace JsonSchema {
+	export type $ = typeof $
+	export type Schema = typeof JsonSchema.Schema.infer
```

Comment:
One issue I foresee here is that generally, if we're publishing this as a standalone package, exporting types like this from a package can be fragile or unresolvable if the package doesn't also directly depend on `arktype`.

Also since I suppose we already have `JsonSchema` types in `arktype`, we'd want to reuse/adapt those as needed. That way, we could just export the types as purely defined in TS.